src/backend/api/Fusion.Resources.Logic/Requests/Commands/ResourceAllocationRequest/Allocation/CanApproveStepHandler.cs

```diff
@@ -1,42 +1,79 @@
+                    IsOrgChartWriteAllowed = true,
+                    IsCreatorAllowed = true
+                },
+                [(AllocationNormalWorkflowV1.SUBTYPE, WorkflowDefinition.PROVISIONING)] = WorkflowAccess.Default,
+
+                [(AllocationJointVentureWorkflowV1.SUBTYPE, AllocationJointVentureWorkflowV1.CREATED)] = WorkflowAccess.Default with
+                {
+                    IsResourceOwnerAllowed = true,
+                    IsParentResourceOwnerAllowed = true,
+                    IsSiblingResourceOwnerAllowed = true,
+                    IsCreatorAllowed = true,
+                },
+                [(AllocationJointVentureWorkflowV1.SUBTYPE, WorkflowDefinition.PROVISIONING)]
+                    = WorkflowAccess.Default,
+
+                [(AllocationEnterpriseWorkflowV1.SUBTYPE, WorkflowDefinition.PROVISIONING)]
+                    = WorkflowAccess.Default,
+
+                [(AllocationDirectWorkflowV1.SUBTYPE, WorkflowDefinition.PROVISIONING)]
+                    = WorkflowAccess.Default,
+            };
             private readonly ResourcesDbContext dbContext;
+            private readonly IHttpContextAccessor httpContextAccessor;
 
-            public CanApproveStepHandler(ResourcesDbContext dbContext)
+            public CanApproveStepHandler(
+                ResourcesDbContext dbContext,
+                IAuthorizationService authService,
+                IHttpContextAccessor httpContextAccessor) : base(authService)
             {
                 this.dbContext = dbContext;
+                this.httpContextAccessor = httpContextAccessor;
             }
 
-            public Task Handle(CanApproveStep notification, CancellationToken cancellationToken)
+            public async Task Handle(CanApproveStep notification, CancellationToken cancellationToken)
             {
-                if (notification.Type != DbInternalRequestType.Allocation)
-                    return Task.CompletedTask;
-
-                // Should be implemented when we need to authorize / validate. 
-                // But at the moment these rules are not set, so would be back and forth to implement
+                if (notification.Type != DbInternalRequestType.Allocation) return;
 
+                var request = await dbContext.ResourceAllocationRequests
+                    .Include(p => p.Project)
+                    .FirstAsync(r => r.Id == notification.RequestId, cancellationToken: cancellationToken);
 
-                //var initiatedBy = await dbContext.Persons.FirstAsync(p => p.Id == notification.InitiatedByDbPersonId);
-                //var request = await dbContext.ResourceAllocationRequests.FirstAsync(r => r.Id == notification.RequestId);
-
-                switch (notification.CurrentStepId)
-                {
-                    case "proposed":
+                var initiator = httpContextAccessor?.HttpContext?.User;
+                if (initiator is null) throw new UnauthorizedWorkflowException();
```

Comment:
Should always include a message when throwing the same exception in different circumstances.